storage/innobase/lock/lock0lock.cc

```diff
@@ -5554,15 +5554,15 @@ lock_sec_rec_read_check_and_lock(
 	there is a probability for lock conflicts between two wsrep
 	high priority threads. To avoid this GAP-locking we mark that
 	this transaction is using unique key scan here. */
-	if (trx->is_wsrep() && wsrep_thd_is_BF(trx->mysql_thd, false))
-		trx->wsrep = 3;
+	if (trx->is_wsrep_BF())
+		trx->wsrep_begin_UK_scan();
```

Comment:
The condition should be part of `trx_t::wsrep_begin_UK_scan()`.

Reply:
Done